content/guides/pedestal-with-component.adoc

```diff
@@ -1,16 +1,419 @@
+a `pedestal.clj` file.
+
+[source, clojure]
+.src/pedestal.clj
+
+----
+include::component/src/pedestal.clj[tags=ns]
+----
+<1> Create a _pedestal_ namespace to house the Pedestal component.
+<2> We need to require `com.stuartsierra.component` to implement the
+`start` and `stop` `Lifecycle` methods.
+<3> We need to require the `io.pedestal.http` for server creation,
+starting and stopping.
+
+Let's start implementing the component.
+----
+include::component/src/pedestal.clj[tags=component-init]
+----
+<1> Create a _Pedestal_ record. This record will contain a number of
+ fields, the first field is _service-map_ which will hold the Pedestal
+ service map component dependency.
+<2> The next field is _start-fn_, this will be passed in through the
+ component constructor and will be used to start the Pedestal
+ server. This function must accept one parameter, the Pedestal service
+ map, and return a Pedestal service map.
+<3> The next field is _stop-fn_, like _start-fn_, the stop function
+ will be passed in through the component constructor. It's responsible
+ for stopping the Pedestal server. This function must accept one
+ parameter, the Pedestal service map, and return a Pedestal service map.
+<4> The _service_ field is created as part of the component
+ initialization process.
+<2> Include the _component/Lifecycle_ protocol since we'll be
+implementing it's methods next.
+
+We'll first implement the `start` method.
+----
+include::component/src/pedestal.clj[tags=component-start]
+----
+<1> The _start_ method contains our component initialization code.
+<2> Our component's initialization process will be idempotent. If
+the _service_ field has already been set we'll return the current
+component.
+<3> The Pedestal service map contains Pedestal's configuration. It is
+a component dependency and will be available during component
+initialization.
+<4> First we'll call the `create-server` function. This will
+initialize the service provider and configure the default
+interceptors.
+<5> Invoke the start function to start the server. As you'll see
+later, this function can simply be `identity` for testing purposes.
+<6> Assoc the initialized service map onto the component as
+`:service`. This is the only state the component needs to manage.
+
+If you've read some of the other guides, this implementation should
+look somewhat familiar. It's a combination of the server-specific code
+used in the link:hello-world[Hello World guide].
+
+Now let's implement the `stop` method.
+----
+include::component/src/pedestal.clj[tags=component-stop]
+----
+<1> The `stop` method contains our component teardown code.
+<2> Like `start`, `stop` will be idempotent. If the component has
+been initialized, we'll pass the initialized service map to the stop
+function, otherwise we won't.
+<3> Return the component with the _service_ field set to `nil`. This
+is preferred over `dissoc` since `dissoc` would return a plain map
```

Comment:
This isn't just preferred, right? If you use dissoc, the component will be broken the next time you try to start it.